### PR TITLE
fix: add [name] selector for all PeopleKit recipient containers (#713)

### PR DESCRIPTION
**Pull request by onestep**
- avoid using closest() if type if specified in options

### CODE REVIEW

File: src/gmail.js [hunk 1]
```diff
@@ -3955,14 +3955,14 @@ var Gmail = function(localJQuery) {
...
-            if( typeof options !== "object" ) options = {};
+            if (typeof options !== "object") options = {};
             const peopleKit = api.check.is_peoplekit_compose(this.$el);
 
             const type_selector = options.type ? "[name=" + options.type + "]" : "";
 
             const found = peopleKit ?
-                this.$el.find("tr.bzf " + type_selector + " div[data-hovercard-id]").map((_, el) => ({
-                    type: el.closest("div[name]").getAttribute("name"),
+                this.$el.find("tr.bzf " + (type_selector || "div[name]") + " div[data-hovercard-id]").map((_, el) => ({
```
Comment (josteink) [inline src/gmail.js:3964]: Wouldn't it be better to make `"div[name]"` the new default for `type_selector`  (line 3961) when no type is provided? Because that's effectively what this change does, right?

It would make this code a little bit easier/cleaner to read.

File: src/gmail.js [hunk 2]
```diff
@@ -3955,14 +3955,14 @@ var Gmail = function(localJQuery) {
...
+                    type: options.type || el.closest("div[name]").getAttribute("name"),
```
Comment (josteink) [inline src/gmail.js:3965]: Could another option here be to reverse this statement, using a null coalescing operator?

`````js
type: el.closest("div[name]")?.getAttribute("name") || options.type,
`````

Or even more: Is this change really needed at all? with the previous fix? :)

File: src/gmail.js (hunk 1 above)
Comment (onestep) [reply src/gmail.js:3964]: I would agree if there would be no `is_peoplekit_compose` check. In other case, there is `<input name="to">`, and this selector is appended as a part of input selector (not as a parent).

File: src/gmail.js (hunk 2 above)
Comment (onestep) [reply src/gmail.js:3965]: This change allows us to avoid traversing the DOM, therefore providing result faster. If we use specific `"div[name=to]"` parent selector, there is no need to look up for the parent, and then taking its `name` attribute. 🙂

Comment (josteink) [reply src/gmail.js:3965]: Fair enough!

File: src/gmail.js (hunk 1 above)
Comment (josteink) [reply src/gmail.js:3964]: Right. I see. Let’s keep this as is too then.

**Comment by josteink** [comment]
Looks good to me. Consider it merged.

**Review by josteink — APPROVED** [review]
